File: projetoJava/src/Professor.java
import java.util.ArrayList;
import java.util.List;

class Professor {
    private String nome;
    private List<Escola> escolas;

    public Professor(String nome) {
        this.nome = nome;
        this.escolas = new ArrayList<>();
    }

    public void adicionarEscola(Escola escola) {
        if (!escolas.contains(escola)) {
            escolas.add(escola);
            escola.adicionarProfessor(this);
        }
    }

    public void exibirEscolas() {
        System.out.println("Professor " + nome + " leciona nas seguintes escolas:");
        for (Escola escola : escolas) {
            System.out.println("- " + escola.getNome());
        }
    }

    public String getNome() {
        return nome;
    }
}

File: projetoGo/professor.go
package main

import "fmt"

type Professor struct {
	nome    string
	escolas []*Escola
}

func (p *Professor) AdicionarEscola(escola *Escola) {
	if !p.temEscola(escola) {
		p.escolas = append(p.escolas, escola)
		escola.AdicionarProfessor(p)
	}
}

func (p *Professor) temEscola(escola *Escola) bool {
	for _, e := range p.escolas {
		if e == escola {
			return true
		}
	}
	return false
}

func (p *Professor) ExibirEscolas() {
	fmt.Printf("Professor %s leciona nas seguintes escolas:\n", p.nome)
	for _, escola := range p.escolas {
		fmt.Printf("- %s\n", escola.nome)
	}
}
